pygmt/tests/test_grdtrack.py

```diff
@@ -1,110 +1,74 @@
-
-
-@pytest.fixture(scope="module", name="ncfile")
-def fixture_ncfile():
+@pytest.fixture(scope="module", name="dataframe")
+def fixture_dataframe():
     """
-    Load the ncfile.
+    Load a pandas DataFrame with points.
     """
-    return which("@earth_relief_01d", download="a")
+    points = [
+        [-51.613, -17.93],
+        [-48.917, -22.434],
+        [-50.444, -16.358],
+        [-50.721, -16.628],
+        [-51.394, -12.196],
+        [-50.207, -18.404],
+        [-52.56, -16.977],
+        [-51.866, -19.794],
+        [-48.001, -14.144],
+        [-54.438, -19.193],
+    ]
+    return pd.DataFrame(data=points, columns=["longitude", "latitude"])
 
 
-def test_grdtrack_input_dataframe_and_dataarray(dataarray, dataframe):
+def test_grdtrack_input_dataframe_and_dataarray(dataarray, dataframe, expected_array):
     """
     Run grdtrack by passing in a pandas.DataFrame and xarray.DataArray as
     inputs.
     """
     output = grdtrack(points=dataframe, grid=dataarray, newcolname="bathymetry")
     assert isinstance(output, pd.DataFrame)
     assert output.columns.to_list() == ["longitude", "latitude", "bathymetry"]
-    npt.assert_allclose(output.iloc[0], [-110.9536, -42.2489, -2797.394987])
-
-    return output
-
-
-def test_grdtrack_input_csvfile_and_dataarray(dataarray, csvfile):
-    """
-    Run grdtrack by passing in a csvfile and xarray.DataArray as inputs.
-    """
-    try:
-        output = grdtrack(points=csvfile, grid=dataarray, outfile=TEMP_TRACK)
-        assert output is None  # check that output is None since outfile is set
-        assert os.path.exists(path=TEMP_TRACK)  # check that outfile exists at path
-
-        track = pd.read_csv(TEMP_TRACK, sep="\t", header=None, comment=">")
-        npt.assert_allclose(track.iloc[0], [-110.9536, -42.2489, -2797.394987])
-    finally:
-        os.remove(path=TEMP_TRACK)
-
-    return output
-
-
-def test_grdtrack_input_dataframe_and_ncfile(dataframe, ncfile):
-    """
-    Run grdtrack by passing in a pandas.DataFrame and netcdf file as inputs.
-    """
-
-    output = grdtrack(points=dataframe, grid=ncfile, newcolname="bathymetry")
-    assert isinstance(output, pd.DataFrame)
-    assert output.columns.to_list() == ["longitude", "latitude", "bathymetry"]
-    npt.assert_allclose(output.iloc[0], [-32.2971, 37.4118, -1939.748245])
-
-    return output
-
-
-def test_grdtrack_input_csvfile_and_ncfile(csvfile, ncfile):
-    """
-    Run grdtrack by passing in a csvfile and netcdf file as inputs.
-    """
-    try:
-        output = grdtrack(points=csvfile, grid=ncfile, outfile=TEMP_TRACK)
-        assert output is None  # check that output is None since outfile is set
-        assert os.path.exists(path=TEMP_TRACK)  # check that outfile exists at path
-
-        track = pd.read_csv(TEMP_TRACK, sep="\t", header=None, comment=">")
-        npt.assert_allclose(track.iloc[0], [-32.2971, 37.4118, -1939.748245])
-    finally:
-        os.remove(path=TEMP_TRACK)
```

Comment:
> @weiji14, as the original author of these tests, what is your opinion on removing the tests than involve file i/o from this module?

I think we should keep these tests. `grdtrack` is one of those rare GMT modules that require both a vector and a raster input, and from memory (#1304 and #1608), there have been instances where the output from a csvfile+ncfile is not the same as a dataframe+dataarray. So my vote would be to keep these as regression tests since they check that all these input combinations produce the same result.